File: App/MainApp/CommandDrivers.vb
Option Strict On
Option Infer Off
Option Explicit On

Imports Autodesk.AutoCAD.ApplicationServices
Imports Autodesk.AutoCAD.Runtime

' This decorator tells the AutoCAD runtime to load the entire CommandDrivers class and look for <CommandMethods> to load as callable from the AutoCAD command line
<Assembly: CommandClass(GetType(CommandDrivers))>
Public Class CommandDrivers
    Public Class Constants
        Public Const GroupName As String = "MainAppCommands"
    End Class


    <CommandMethod(Constants.GroupName, "MyCommand", CommandFlags.Session)>
    Public Shared Sub CommandMethod()
        Dim doc As Document = Application.DocumentManager.MdiActiveDocument
        doc.Editor.WriteMessage("Hello from the command drivers class" + vbLf)
        BusinessLogic.Commands.Example.Run()
    End Sub

End Class

File: CSharpApp/MainApp/CommandDrivers.cs
using Autodesk.AutoCAD.Runtime;
using System.Threading.Tasks;

// This decorator tells the AutoCAD runtime to load the entire CommandDrivers class and look for <CommandMethods> to load as callable from the AutoCAD command line
[assembly: CommandClass(typeof(MainApp.CommandDrivers))]

namespace MainApp
{
    public class CommandDrivers
    {
        public class Constants
        {
            public const string GroupName = "MainAppCommands";
        }


        [CommandMethod(Constants.GroupName, "MyCommand", CommandFlags.Session)]
        public static void CommandMethod()
        {
            var doc = Autodesk.AutoCAD.ApplicationServices.Core.Application.DocumentManager.MdiActiveDocument;
            doc.Editor.WriteMessage("Hello from the command drivers class" + Microsoft.VisualBasic.Constants.vbLf);
            BusinessLogic.Commands.Example.Run();
        }

        [CommandMethod(Constants.GroupName, "MyTests", CommandFlags.Session)]
        public static async Task MyTestsAsync()
        {
            var doc = Autodesk.AutoCAD.ApplicationServices.Core.Application.DocumentManager.MdiActiveDocument;

            var allRegressionTests = typeof(RegressionTests.RegressionTests);
            var allIntegrationTests = typeof(IntegrationTests.Tests);
            var allRegressionTestTypes = new[] { allRegressionTests, allIntegrationTests };

            var workflow = new Workflows.NUnitTestRunner();
            await workflow.Run(allRegressionTestTypes);

        }
    }
}